test_build_deps_on_distutils[packaging] failing in CI

When I run the tests locally, they pass, but in CI, the test is failing thus:

```
___________________ test_build_deps_on_distutils[packaging] ____________________

request = <FixtureRequest for <Function test_build_deps_on_distutils[packaging]>>
tmpdir_factory = TempdirFactory(_tmppath_factory=TempPathFactory(_given_basetemp=None, _trace=<pluggy._tracing.TagTracerSub object at 0x7f7e5e63a3d0>, _basetemp=PosixPath('/tmp/pytest-of-runner/pytest-0')))
build_dep = 'packaging'

    @pytest.mark.parametrize("build_dep", build_deps)
    @pytest.mark.skipif(
        sys.version_info < (3, 6), reason='run only on late versions')
    def test_build_deps_on_distutils(request, tmpdir_factory, build_dep):
        """
        All setuptools build dependencies must build without
        setuptools.
        """
        if 'pyparsing' in build_dep:
            pytest.xfail(reason="Project imports setuptools unconditionally")
        build_target = tmpdir_factory.mktemp('source')
        build_dir = download_and_extract(request, build_dep, build_target)
        install_target = tmpdir_factory.mktemp('target')
>       output = install(build_dir, install_target)

setuptools/tests/test_integration.py:138: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
setuptools/tests/test_integration.py:157: in install
    cmd, cwd=pkg_dir, env=env, stderr=subprocess.STDOUT)
/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/subprocess.py:411: in check_output
    **kwargs).stdout
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

input = None, capture_output = False, timeout = None, check = True
popenargs = (['/home/runner/work/setuptools/setuptools/.tox/python/bin/python', 'setup.py', 'install', '--prefix', '/tmp/pytest-of-runner/pytest-0/target1'],)
kwargs = {'cwd': '/tmp/pytest-of-runner/pytest-0/source1/packaging-20.7', 'env': {'CI': 'true', 'COVERAGE_FILE': '/home/runner/...'COV_CORE_DATAFILE': '/home/runner/work/setuptools/setuptools/.tox/.coverage.python', ...}, 'stderr': -2, 'stdout': -1}
process = <subprocess.Popen object at 0x7f7e40c25110>
stdout = b"/home/runner/work/setuptools/setuptools/.tox/python/bin/python: can't open file 'setup.py': [Errno 2] No such file or directory\n"
stderr = None, retcode = 2

    def run(*popenargs,
            input=None, capture_output=False, timeout=None, check=False, **kwargs):
        """Run command with arguments and return a CompletedProcess instance.
    
        The returned instance will have attributes args, returncode, stdout and
        stderr. By default, stdout and stderr are not captured, and those attributes
        will be None. Pass stdout=PIPE and/or stderr=PIPE in order to capture them.
    
        If check is True and the exit code was non-zero, it raises a
        CalledProcessError. The CalledProcessError object will have the return code
        in the returncode attribute, and output & stderr attributes if those streams
        were captured.
    
        If timeout is given, and the process takes too long, a TimeoutExpired
        exception will be raised.
    
        There is an optional argument "input", allowing you to
        pass bytes or a string to the subprocess's stdin.  If you use this argument
        you may not also use the Popen constructor's "stdin" argument, as
        it will be used internally.
    
        By default, all communication is in bytes, and therefore any "input" should
        be bytes, and the stdout and stderr will be bytes. If in text mode, any
        "input" should be a string, and stdout and stderr will be strings decoded
        according to locale encoding, or by "encoding" if set. Text mode is
        triggered by setting any of text, encoding, errors or universal_newlines.
    
        The other arguments are the same as for the Popen constructor.
        """
        if input is not None:
            if kwargs.get('stdin') is not None:
                raise ValueError('stdin and input arguments may not both be used.')
            kwargs['stdin'] = PIPE
    
        if capture_output:
            if kwargs.get('stdout') is not None or kwargs.get('stderr') is not None:
                raise ValueError('stdout and stderr arguments may not be used '
                                 'with capture_output.')
            kwargs['stdout'] = PIPE
            kwargs['stderr'] = PIPE
    
        with Popen(*popenargs, **kwargs) as process:
            try:
                stdout, stderr = process.communicate(input, timeout=timeout)
            except TimeoutExpired as exc:
                process.kill()
                if _mswindows:
                    # Windows accumulates the output in a single blocking
                    # read() call run on child threads, with the timeout
                    # being done in a join() on those threads.  communicate()
                    # _after_ kill() is required to collect that and add it
                    # to the exception.
                    exc.stdout, exc.stderr = process.communicate()
                else:
                    # POSIX _communicate already populated the output so
                    # far into the TimeoutExpired exception.
                    process.wait()
                raise
            except:  # Including KeyboardInterrupt, communicate handled that.
                process.kill()
                # We don't call process.wait() as .__exit__ does that for us.
                raise
            retcode = process.poll()
            if check and retcode:
                raise CalledProcessError(retcode, process.args,
>                                        output=stdout, stderr=stderr)
E               subprocess.CalledProcessError: Command '['/home/runner/work/setuptools/setuptools/.tox/python/bin/python', 'setup.py', 'install', '--prefix', '/tmp/pytest-of-runner/pytest-0/target1']' returned non-zero exit status 2.

/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/subprocess.py:512: CalledProcessError
```

--- a/changelog.d/2471.misc.rst
+++ b/changelog.d/2471.misc.rst
@@ -1,0 +1,1 @@
+Removed the tests that guarantee that the vendored dependencies can be built by distutils.

--- a/setuptools/tests/test_integration.py
+++ b/setuptools/tests/test_integration.py
@@ -6,11 +6,6 @@ Try to install a few packages.
 import glob
 import os
 import sys
-import re
-import subprocess
-import functools
-import tarfile
-import zipfile
 import urllib.request
 
 import pytest
@@ -117,54 +112,3 @@ def test_pyuri(install_context):
 
     # The package data should be installed.
     assert os.path.exists(os.path.join(pyuri.location, 'pyuri', 'uri.regex'))
-
-
-build_deps = ['appdirs', 'packaging', 'pyparsing', 'six']
-
-
-@pytest.mark.parametrize("build_dep", build_deps)
-def test_build_deps_on_distutils(request, tmpdir_factory, build_dep):
-    """
-    All setuptools build dependencies must build without
-    setuptools.
-    """
-    if 'pyparsing' in build_dep:
-        pytest.xfail(reason="Project imports setuptools unconditionally")
-    build_target = tmpdir_factory.mktemp('source')
-    build_dir = download_and_extract(request, build_dep, build_target)
-    install_target = tmpdir_factory.mktemp('target')
-    output = install(build_dir, install_target)
-    for line in output.splitlines():
-        match = re.search('Unknown distribution option: (.*)', line)
-        allowed_unknowns = [
-            'test_suite',
-            'tests_require',
-            'python_requires',
-            'install_requires',
-            'long_description_content_type',
-        ]
-        assert not match or match.group(1).strip('"\'') in allowed_unknowns
-
-
-def install(pkg_dir, install_dir):
-    with open(os.path.join(pkg_dir, 'setuptools.py'), 'w') as breaker:
-        breaker.write('raise ImportError()')
-    cmd = [sys.executable, 'setup.py', 'install', '--prefix', str(install_dir)]
-    env = dict(os.environ, PYTHONPATH=str(pkg_dir))
-    output = subprocess.check_output(
-        cmd, cwd=pkg_dir, env=env, stderr=subprocess.STDOUT)
-    return output.decode('utf-8')
-
-
-def download_and_extract(request, req, target):
-    cmd = [
-        sys.executable, '-m', 'pip', 'download', '--no-deps',
-        '--no-binary', ':all:', req,
-    ]
-    output = subprocess.check_output(cmd, encoding='utf-8')
-    filename = re.search('Saved (.*)', output).group(1)
-    request.addfinalizer(functools.partial(os.remove, filename))
-    opener = zipfile.ZipFile if filename.endswith('.zip') else tarfile.open
-    with opener(filename) as archive:
-        archive.extractall(target)
-    return os.path.join(target, os.listdir(target)[0])

